server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -1498,7 +1498,11 @@ public static int parseInt(String s, int minValue, int maxValue, String key, boo
         return value;
     }
 
-    static long parseLong(String s, long minValue, String key, boolean isFiltered) {
+    public static long parseLong(String s, long minValue, String key) {
+        return parseLong(s, minValue, key);
```

Comment:
This looks like accidental recursion, should it be `parseLong(s, minValue, key, true|false);` ?